com.unity.netcode.gameobjects/Runtime/Messaging/Messages/SnapshotDataMessage.cs

```diff
@@ -1,166 +1,55 @@
-using System;
 using Unity.Collections;
-using Unity.Collections.LowLevel.Unsafe;
 using UnityEngine;
 
 namespace Unity.Netcode
 {
     internal struct SnapshotDataMessage : INetworkMessage
     {
-        public int CurrentTick;
-        public ushort Sequence;
+        internal FastBufferWriter WriteBuffer;
+        internal FastBufferReader ReadBuffer;
+        private int m_BufferSize;
 
-        public ushort Range;
-
-        public byte[] SendMainBuffer;
-        public NativeArray<byte> ReceiveMainBuffer;
-
-        public struct AckData
+        // a constructor with an unused parameter is used because C# doesn't allow parameter-less constructors
+        public SnapshotDataMessage(int unused)
         {
-            public ushort LastReceivedSequence;
-            public ushort ReceivedSequenceMask;
+            m_BufferSize = 10000;
```

Comment:
this to-be-refactored buffer size might be the unused param

Reply:
I'd suggest using a growing buffer here, maybe starting at 1024 with int.MaxValue as the max. (FastBufferWriter becoes a growing buffer if you provide a third parameter.)

Reply:
+1 and also maybe inject `m_MaxBufferSize` via the struct constructor instead of `int unused` there :)